### PR TITLE
fix(tool): codecov carry forward flags

### PR DESCRIPTION
**Pull request by Leptopoda**
tested in: https://github.com/Leptopoda/nextcloud-neon/pull/4

I'll force run the CI for all packages to check it still works over here.

### CODE REVIEW

File: tool/upload_coverage.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# DO NOT enable -x as it will leak the `CODECOV_TOKEN` secret
+set -euo pipefail
+cd "$(dirname "$0")/.."
+
+if [ ! -v GITHUB_REPOSITORY ]; then
+    exit
+fi
```
Comment (provokateurin) [inline tool/upload_coverage.sh:8]:
```suggestion
if [ ! -v GITHUB_REPOSITORY ]; then
    exit 1
fi
```
Always indicate that an error happened.

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: tool/upload_coverage.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+if [ ! -f codecov ]; then
+    curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import
+    curl -Os https://uploader.codecov.io/latest/linux/codecov
+    curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM
+    curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM.sig
+    gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
+
+    shasum -a 256 -c codecov.SHA256SUM
+    chmod +x codecov
```
[the remark this answers is not in the document]
Comment (Leptopoda) [reply tool/upload_coverage.sh:18]: I'm not aware of any packaged version of the cli for ubuntu.
I just followed the codecov docs for "linux":
https://docs.codecov.com/docs/codecov-uploader#integrity-checking-the-codecov-cli

[the remark this answers is not in the document]
Comment (provokateurin) [reply tool/upload_coverage.sh]: Hm ok :woman_shrugging:

File: tool/upload-coverage.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+if [ ! -f codecov ]; then
+    curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import
+    curl -Os https://uploader.codecov.io/latest/linux/codecov
+    curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM
+    curl -Os https://uploader.codecov.io/latest/linux/codecov.SHA256SUM.sig
+    gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
+
+    shasum -a 256 -c codecov.SHA256SUM
+    chmod +x codecov
+fi
```
Comment (provokateurin) [inline tool/upload-coverage.sh:19]: The install step should happen in the CI workflow and not in the script.

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Comment by Leptopoda** [comment]
Ok this approach does still work :tada:
I'll change the last complaints shortly

File: .github/workflows/dart.yml [hunk 4]
```diff
@@ -41,18 +41,24 @@ jobs:
...
+        run: melos test
+
+      - name: Setup Codecov
         run: |
-          mapfile -t packages < <(melos list --parsable --diff=${{ github.event.pull_request.head.sha }}~${{ github.event.pull_request.commits }}..${{ github.event.pull_request.head.sha }} --include-dependents | xargs -r -n1 basename)
-          if [ -z "$packages" ]; then exit 0; fi
-          packages_glob="$(printf ",%s" "${packages[@]}")"
-          packages_glob="${packages_glob:1}"
-          MELOS_PACKAGES="$packages_glob" melos test
+          cd /tmp
```
Comment (provokateurin) [inline .github/workflows/dart.yml:48]: You could just create the bin dir directly and cd into it instead of moving the binary afterwards.

Comment (Leptopoda) [reply .github/workflows/dart.yml:48]: But then the key and checksum would also end up in our PATH which I want to avoid

Comment (provokateurin) [reply .github/workflows/dart.yml:48]: Ah right, that makes sense :+1:

File: .github/workflows/dart.yml [hunk 5]
```diff
@@ -41,18 +41,25 @@ jobs:
...
-          codecov_yml_path: ./github/codecov.yaml
-          exclude: .fvm
-          fail_ci_if_error: true
-          slug: nextcloud/neon
+        run: |
+          melos exec --file-exists="coverage/lcov.info" --concurrency=1 -- "
+            codecov --verbose upload-process --fail-on-error -F "$MELOS_PACKAGE_NAME" -f "$MELOS_PACKAGE_PATH/coverage/lcov.info" -t "${{ secrets.CODECOV_TOKEN }}"
+          "
+        env:
+          CODECOV_TOKEN: "$PATH:/tmp/bin"
```
Comment (provokateurin) [inline .github/workflows/dart.yml:65]: This makes no sense?

Comment (Leptopoda) [reply .github/workflows/dart.yml:65]: :sweat_smile:

**Pull request merged**